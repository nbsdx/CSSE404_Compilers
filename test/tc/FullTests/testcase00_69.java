/*
checks method overloading 
*/
class Test {
    public static void main(String[] args) {
	int x = 5;
	FooExt fe = new FooExt();
	boolean val = fe.predicate(2, 4);
    }
}


class Foo {
    public boolean predicate(int y) {
	return false;
    }
}

class FooExt extends Foo {
    //ok...
    public boolean predicate(int z) {
	return (z > 0);
    }
    //not ok...
    public boolean predicate(int y, int z) {
	return (y < z);
    }
}
