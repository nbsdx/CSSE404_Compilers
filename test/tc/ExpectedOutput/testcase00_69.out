Cannot redeclare method predicate
Cannot overload methods.  Method predicate has different type signature than inherited method of the same name.
